src/components/workers-page/shifts-tab/shifts-tab.component.tsx

```diff
@@ -1,5 +1,110 @@
-import React from "react";
+import React, { useState } from "react";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableContainer from "@material-ui/core/TableContainer";
+import TableRow from "@material-ui/core/TableRow";
+import {
+  DisplayedShiftData,
+  ShiftCode,
+  ShiftHours,
+  ShiftName,
+} from "../../../common-models/shift-info.model";
+import { Button } from "../../common-components";
+import { createStyles, makeStyles } from "@material-ui/core/styles";
+import { EnhancedTableHeaderComponent } from "./enhanced-table-header.component";
+import ScssVars from "../../../assets/styles/styles/custom/_variables.module.scss";
+import ShiftDrawerComponent, { ShiftDrawerMode } from "./shift-drawer.component";
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    root: {
+      paddingTop: 0,
+      width: "100%",
+    },
+    tableCell: {
+      color: ScssVars.primary,
+      fontWeight: "normal",
+      fontSize: ScssVars.fontSizeBase,
+      fontFamily: ScssVars.fontFamilyPrimary,
+      letterSpacing: ScssVars.headingLetterSpacing,
+    },
+    row: {
+      borderTop: `2px solid ${ScssVars.workerTableBorderColor}`,
+    },
+    colorSample: {
+      width: "18px",
+      height: "18px",
+      borderRadius: "24px",
+    },
+  })
+);
 
 export default function ShiftTab(): JSX.Element {
-  return <h2>Zmiany</h2>;
+  const classes = useStyles();
+  const [open, setIsOpen] = useState(false);
+  const [mode, setMode] = useState(ShiftDrawerMode.ADD_NEW);
+  const shiftData: DisplayedShiftData[] = [
+    { code: ShiftCode.D, name: ShiftName.D, hours: ShiftHours.D, color: "CBEECB" },
+    { code: ShiftCode.N, name: ShiftName.N, hours: ShiftHours.N, color: "B7BCC7" },
+    { code: ShiftCode.R, name: ShiftName.R, hours: ShiftHours.R, color: "FFE880" },
+    { code: ShiftCode.P, name: ShiftName.P, hours: ShiftHours.P, color: "B3E3FF" },
+    { code: ShiftCode.L4, name: ShiftName.L4, hours: ShiftHours.L4, color: "EEB3B3" },
+    { code: ShiftCode.U, name: ShiftName.U, hours: ShiftHours.U, color: "FFDBC3" },
```

Comment:
to nie powinno być tutaj, tylko [*gdzie??*]

Reply:
Widziałem, że to już jest przeniesione do struktury i teraz jest super.